conda_libmamba_solver/utils.py

```diff
@@ -31,3 +36,14 @@ def escape_channel_url(channel):
         parts = parts.replace(path=path)
         return str(parts)
     return channel
+
+
+@lru_cache(maxsize=None)
+def is_channel_available(channel_url) -> bool:
+    if context.offline:
+        # We don't know where the channel might be (even file:// might be a network share)
+        # so we play it safe and assume it's not available
+        return False
+    if channel_url.startswith("file://"):
+        return Path(url_to_path(channel_url)).is_dir()
+    return requests.head(f"{channel_url}/noarch/repodata.json").ok
```

Comment:
Could we use conda's session object for that?

This doesn't handle exceptions as well, right?